cpp/src/lists/contains.cu

```diff
@@ -124,14 +126,15 @@ struct lookup_functor {
           return;
         }
 
-        auto search_key    = search_key_and_validity.first;
-        d_bools[row_index] = thrust::find_if(thrust::seq,
-                                             list.pair_begin<ElementType>(),
-                                             list.pair_end<ElementType>(),
-                                             [search_key] __device__(auto element_and_validity) {
-                                               return element_and_validity.second &&
-                                                      (element_and_validity.first == search_key);
-                                             }) != list.pair_end<ElementType>();
+        auto search_key = search_key_and_validity.first;
+        d_bools[row_index] =
+          thrust::find_if(thrust::seq,
+                          list.pair_begin<ElementType>(),
+                          list.pair_end<ElementType>(),
+                          [search_key] __device__(auto element_and_validity) {
+                            return element_and_validity.second &&
+                                   equality_compare(element_and_validity.first, search_key);
```

Comment:
By the by, invoking `equals_compare()` in the case of a `scalar` search-key is not ideal. We should be able to check whether the search-key is both scalar and `NaN` **once**, and not in every iteration. 

I'm working on optimizing that now.

Reply:
That sounds a little like early optimization. It'd be like special casing if `scalar` is 0 or any other value.

Reply:
Ah, I hadn't thought of it like that. I was looking at `NaN` in a similar way to `null`. 
Thank you for the guidance. I'll let this be.